src/main/java/com/deque/axe/android/rules/hierarchy/SwitchName.java

```diff
@@ -1,11 +1,33 @@
 package com.deque.axe.android.rules.hierarchy;
 
+import com.deque.axe.android.AxeView;
 import com.deque.axe.android.constants.AndroidClassNames;
+import com.deque.axe.android.constants.AxeStatus;
 import com.deque.axe.android.rules.hierarchy.base.ModifiableViewName;
+import com.deque.axe.android.utils.AxeTextUtils;
+import com.deque.axe.android.wrappers.AxeProps;
 
 public class SwitchName extends ModifiableViewName {
 
   public SwitchName() {
     super(AndroidClassNames.SWITCH);
   }
-}
+
+  @Override
+  public void collectProps(AxeView axeView, AxeProps axeProps) {
+    axeProps.put(AxeProps.Name.VISIBLE_TEXT, axeView.text);
+
+    super.collectProps(axeView, axeProps);
+  }
+
+  @Override
+  public String runRule(AxeProps axeProps) {
+    final String visibleText = axeProps.get(AxeProps.Name.VISIBLE_TEXT, String.class);
+
+    if (!AxeTextUtils.isNullOrEmpty(visibleText)
+            && !visibleText.equalsIgnoreCase("on") && !visibleText.equalsIgnoreCase("off")) {
```

Comment:
Extract this into a method with a semantic name that clarifies, in perpetuity, _why_ this is valid.

Reply:
I think this belongs in AxeView. I would propose adding a method that says `containsNonStateText()` or `hasNameText()` I'm having trouble naming this method.

Reply:
Agreed. How about `hasImmutableText()` or `hasOperatingSystemText()` ?